packages/server/src/integrations/oracle.ts

```diff
@@ -322,6 +323,30 @@ class OracleIntegration extends Sql implements DatasourcePlus {
     this.schemaErrors = final.errors
   }
 
+  async testConnection() {
+    const response: ConnectionInfo = {
+      connected: false,
+    }
+    let connection
+    try {
+      connection = await this.getConnection()
+      response.connected = true
+    } catch (err: any) {
+      response.connected = false
+      response.error = err.message
+    } finally {
+      if (connection) {
+        try {
+          await connection.close()
+        } catch (err: any) {
```

Comment:
Why is this? Should the connected:`true/false` not be already setup in the catch above?

Reply:
It seems some Oracle issues don't crop up until you actually try to close the connection - doing this means we safely catch all errors that could occur (start and end of connection).